examples/Remarketing/UpdateAudienceTargetRestriction.php

```diff
@@ -0,0 +1,241 @@
+{
+    private const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    private const AD_GROUP_ID = 'INSERT_AD_GROUP_ID_HERE';
+
+    private const PAGE_SIZE = 1000;
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT,
+            ArgumentNames::AD_GROUP_ID => GetOpt::REQUIRED_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID,
+                $options[ArgumentNames::AD_GROUP_ID] ?: self::AD_GROUP_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+            exit(1);
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+            exit(1);
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ID of the ad group to update
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId
+    ) {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates a query that retrieves the targeting settings from a given ad group.
+        $query = "SELECT ad_group.id, ad_group.name, " .
+            "ad_group.targeting_setting.target_restrictions " .
+            "FROM ad_group " .
+            "WHERE ad_group.id = $adGroupId";
+
+        // Issues a search request.
+        $response =
```

Comment:
Is this intended to use search, not searchStream?

Reply:
I replicated the [golden](https://github.com/googleads/google-ads-java/blob/master/google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UpdateAudienceTargetRestriction.java#L119) which uses `GAS.Search` but I am thinking that we may want to switch to `GAS.SearchStreaming` as well.

@devchas any thoughts?

Reply:
No reason in particular. Is there a reason we should favor searchStream in this case?

Reply:
Not really, I think @fiboknacky 's comment was mainly about the fact that we [prefer](https://developers.google.com/google-ads/api/docs/concepts/retrieving-objects) `GAS.SearchStream` since it was released.

@fiboknacky I think this is nit, are you OK to keep this as-is and discuss this further internally?

Reply:
Yes, let's discuss internally.

Reply:
SG, thanks.